modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/ExecutionServiceImpl.java

```diff
@@ -472,6 +464,18 @@ protected AffinityTopologyVersion topologyVersion() {
         return exchangeManager().readyAffinityVersion();
     }
 
+    /** */
+    private static SqlNodeList parse(String qry) {
```

Comment:
lets get rid of org.apache.ignite.internal.processors.query.calcite.prepare.IgnitePlanner#parse ? Additionally i suggest to move this static method into more suitable class ?

Reply:
Good idea, but we cannot get rid of it, because this methods are defined in the Planner interface from Calcite